src/components/FeatureViews/Courses/NewAnnoucementsModal.js

```diff
@@ -0,0 +1,169 @@
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogContentText from "@material-ui/core/DialogContentText";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import DoneIcon from "@material-ui/icons/CheckCircleOutlined";
+import EditIcon from "@material-ui/icons/EditOutlined";
+import Grid from "@material-ui/core/Grid";
+import InputBase from "@material-ui/core/InputBase";
+import Loading from "components/OmouComponents/Loading";
+import LoadingError from "../Accounts/TabComponents/LoadingCourseError";
+import NotificationIcon from "@material-ui/icons/NotificationImportant";
+import Paper from "@material-ui/core/Paper";
+import TextField from "@material-ui/core/TextField";
+import Tooltip from "@material-ui/core/Tooltip";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles } from "@material-ui/core/styles";
+import gql from "graphql-tag";
+import { useMutation } from "@apollo/react-hooks";
+
+const useStyles = makeStyles((theme) => ({}));
+
+const NewAnnoucementsModal = ({ handleClose, open, subject, body, userId }) => {
+  const classes = useStyles();
+  const [sendEmailCheckbox, setSendEmailCheckbox] = useState(false);
+  const [annoucementBody, setAnnoucementBody] = useState(body);
+  const [annoucementSubject, setAnnoucementSubject] = useState(subject);
+  const id = useParams();
+  const user_id = userId.results[0].user.id
+
+//   useEffect(() => {
+//       setAnnoucementBody(body);
+//       setAnnoucementSubject(subject);
+//   },[annoucementBody, annoucementSubject]);
+
+  const CREATE_ANNOUCEMENTS = gql`
+    mutation CreateAnnoucement(
+      $subject: Subject!,
+      $body: Body!,
+      $course_id: Course_Id!,
+      $user_id: User_Id!
+    ) {
+      __typename
+      createAnnoucement(
+        body: $body,
+        course_id: $course_id,
+        subject: $subject,
+        user_id: $user_id
+      ) {
+        created
+        annoucement {
+            body
+            id
+            course {
+              courseId
+            }
+            subject
+            user {
+              id
+            }
+          }
+      }
+    }
+  `;
+
+  //   mutation CreateReviewForEpisode($ep: Episode!, $review: ReviewInput!) {
+  //     createReview(episode: $ep, review: $review) {
+  //       stars
+  //       commentary
+  //     }
+  //   }
+
+  const [createAnnoucement, createAnnoucementResult] = useMutation(
+    CREATE_ANNOUCEMENTS
+  );
+
+  const handleCloseForm = () => {
+    handleClose(false);
+  };
+
+  const handleSubjectChange = useCallback((event) => {
+    setAnnoucementSubject(event.target.value);
+}, []);
+
+const handleBodyChange = useCallback((event) => {
+    setAnnoucementBody(event.target.value);
+}, []);
+
+console.log(annoucementBody)
+console.log(annoucementSubject)
+console.log(user_id)
+console.log(id.id)
+
+  const handlePostForm = async (event) => {
```

Comment:
use apollo to do the heavy lifting of error handling + awaiting the completion of this request. Here's the [apollo mutation documentation](https://www.apollographql.com/docs/react/data/mutations/) in this function, just have
```
const handlePostForm = (event) => createAnnouncement({...});
```
But at line 80 when you create your mutation, specify in the configuration how you want to handle errors/on complete. What your mutation variable should look like:
```
  const [createAnnoucement, createAnnoucementResult] = useMutation(
    CREATE_ANNOUCEMENTS,
    {
       onCompleted: () => handleClose(false),
       onError: (err) => console.log(err)
     }
  );
```